spotbugs/src/main/java/edu/umd/cs/findbugs/detect/StringConcatenation.java

```diff
@@ -161,26 +176,30 @@ public void sawOpcode(int seen) {
             if ((seen == Const.NEW) && getClassConstantOperand().startsWith("java/lang/StringBu")) {
                 state = SEEN_NEW;
                 createPC = getPC();
+            } else if (seen == Const.INVOKEDYNAMIC && "makeConcatWithConstants".equals(getNameConstantOperand())) {
+                state = CONSTRUCTED_STRING_ON_STACK;
+                stringSource = getRegisterOnStack(1);
+                createPC = getPC();
             }
             break;
 
         case SEEN_NEW:
             if (seen == Const.INVOKESPECIAL && Const.CONSTRUCTOR_NAME.equals(getNameConstantOperand())
                     && "(Ljava/lang/String;)V".equals(getSigConstantOperand())
-                    && getClassConstantOperand().startsWith("java/lang/StringBu") && registerOnStack >= 0) {
+                    && getClassConstantOperand().startsWith("java/lang/StringBu") && getRegisterOnStack() >= 0) {
```

Comment:
Perhaps this should be extracted to a local variable, since it's referenced many times.